### ISSUE TITLE
nodeagent failing to start when using citadel in 1.2.5

### ISSUE THREAD

**Issue by nixgadget** [istio/istio#17108]
*Istio*: 1.2.5

I have configured `nodeagent` with `citadel` as the CA provider. However it is failing to start with the following error.

```
2019-09-16T10:13:53.418878Z	info	curl -k -v -XGET  -H "Accept: application/json, */*" -H "User-Agent: node_agent_k8s/v0.0.0 (linux/amd64) kubernetes/$Format" 'https://10.0.0.1:443/api/v1/namespaces/istio-system/configmaps/istio-security'
2019-09-16T10:13:53.427166Z	info	GET https://10.0.0.1:443/api/v1/namespaces/istio-system/configmaps/istio-security 404 Not Found in 8 milliseconds
2019-09-16T10:13:53.427385Z	info	Response Headers:
2019-09-16T10:13:53.427488Z	info	    Audit-Id: fe6cee0a-5020-4c2e-aaef-0302ac0a8358
2019-09-16T10:13:53.427614Z	info	    Content-Type: application/json
2019-09-16T10:13:53.427818Z	info	    Content-Length: 208
2019-09-16T10:13:53.427910Z	info	    Date: Mon, 16 Sep 2019 10:13:53 GMT
```

This is because I have deployed Istio to a custom namespace and not `istio-system`.
Looking at the code [client.go#L36](https://github.com/istio/istio/blob/1.2.5/security/pkg/nodeagent/caclient/client.go#L36) it turns out `istio-system` is hard-coded.
Is there anyway to make this configurable ?

**Comment by pitlv2109** [comment]
I think we fixed it here https://github.com/istio/istio/pull/15199 but didn't cherry-pick it to 1.2.x :(

**Comment by pitlv2109** [comment]
We will most likely have this ready for 1.2.7 (but 1.3 already has it)

**Comment by nixgadget** [comment]
@pitlv2109 thanks for the update. So are you saying even though it was labelled it was never cherry picked into 1.2 branch ? Any guarantees it will go into 1.2.7 release ? 
If so we can close this ticket.

**Comment by howardjohn** [comment]
@pitlv2109 did you cherrypick to 1.2 yet?

**Comment by pitlv2109** [comment]
I created a cherry-pick this morning https://github.com/istio/istio/pull/17316, but Istio 1.3 should fix your issue, @nixgadget.
 
@fpesce possible for 1.2.7?

**Comment by howardjohn** [comment]
Closing as cherry pick merged, will be in next 1.2 release
